pkg/api/controllers/base_controller.go

```diff
@@ -53,6 +55,8 @@ func coreErrorToErrorCode(err error) (int, string) {
 		return http.StatusBadRequest, ErrInsufficientFund
 	case ledger.IsValidationError(err):
 		return http.StatusBadRequest, ErrValidation
+	case strings.HasPrefix(err.Error(), querystrings.ErrInvalidQueryValue):
```

Comment:
We should use types for errors instead of string comparison.

Reply:
Ok, I changed it to use the `ledger.ValidationError` type. Is it what you meant?